lib/config.js

```diff
@@ -14,29 +21,11 @@ exports.defaultConfigFile = defaultConfigFile;
 exports.read = function( configFile, callback ) {
 	if ( typeof configFile === 'function' ) {
 		callback = configFile;
-		configFile = defaultConfigFile;
-	}
-
-	if ( !configFile ) {
-		configFile = defaultConfigFile;
+		configFile = null;
 	}
 
-	var configDir = path.dirname( configFile );
-
-	mkdirp( configDir, function( err ) {
-		if ( err ) {
-			return callback( err );
-		}
-
-		fs.exists( configFile, function( exists ) {
-			if ( exists ) {
-				fs.readFile( configFile, function( err, src ) {
-					callback( err, JSON.parse( src ), configDir );
-				} );
-			} else {
-				callback( err, null, configDir );
-			}
-		} );
```

Comment:
@gregpabian Safe to remove this entire block right? It makes no sense to create a directory on read, `fs.exists` is deprecated, and mkdir isn't even necessary (because both `configModule.write` and `createProfiles` already do this themselves). And the third argument to the callback (`configDir`) is never used.